### PR TITLE
Fix non-lclVar odd-byte struct passing

### LINKED ISSUE #19288: Incorrect up-sizing for non-power-of-2 structs that are not LclVars
When loading a struct from an arbitrary location, it is not safe to load any more bytes than the specified size of the struct. For instance, in the case where you're loading a 3-byte struct that's in the last three bytes of a page, the next byte following the struct may be on a page that is unmapped or not readable.

A JIT dump showed this transformation during the morph process (note the `OBJ(3)` turns into an `IND int`):
```
fgMorphTree BB02, stmt 23 (before)
               [000124] --CXG-------              *  CALL      void   Test.Foo
               [000121] ------------ this in x0   +--*  LCL_VAR   ref    V00 this         
               [000123] ---XG------- arg1         \--*  OBJ(3)    struct
               [000122] ------------                 \--*  LCL_VAR   long   V05 loc4         
argSlots=2, preallocatedArgCount=0, nextSlotNum=0, outgoingArgSpaceSize=0

Sorting the arguments:
Deferred argument ('x1'):
               [000123] ---XG+------              *  IND       int   
               [000122] -----+------              \--*  LCL_VAR   long   V05 loc4         
Replaced with placeholder node:
               [000186] ----------L-              *  ARGPLACE  int   
Deferred argument ('x0'):
               [000121] -----+------              *  LCL_VAR   ref    V00 this         
Replaced with placeholder node:
               [000188] ----------L-              *  ARGPLACE  ref   
```

Debugging `fgMorphArgs` shows that we’re taking this codepath which is changing the size from 3 to 4:

```c++
#if defined(_TARGET_ARM64_) || defined(UNIX_AMD64_ABI)
                    // For ARM64 or AMD64/UX we can pass non-power-of-2 structs in a register.
                    if ((howToPassStruct == SPK_PrimitiveType) && // Passed in a single register
                        !isPow2(originalSize))                    // size is 3,5,6 or 7 bytes
                    {
                        originalSize = genTypeSize(structBaseType);
                    }
#endif //  _TARGET_ARM64_ || UNIX_AMD64_ABI
```

This branch was modified a couple of months ago in https://github.com/dotnet/coreclr/pull/18358 to remove an additional check in there that was also checking if the source was a LclVar before adjusting the size.

@CarolEidt Can you please take a look and determine whether we need to reintroduce that check or make a better fix?

### PR DESCRIPTION
**Pull request by CarolEidt**
When a non-power-of-2 struct is passed, we can only use an IND if the arg is a local.

Fix #19288

### CODE REVIEW

File: tests/src/JIT/Regression/JitBlue/GitHub_19288/GitHub_19288.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// This test was extracted from the corefx System.Numerics.Vectors tests,
```
Comment (adityamandaleeka) [inline tests/src/JIT/Regression/JitBlue/GitHub_19288/GitHub_19288.cs]: Was this comment copied from another test?

File: tests/src/JIT/Regression/JitBlue/GitHub_19288/GitHub_19288.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            try
+            {
+                new MyClass().DoStuff();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("Failed with exception: " + e.Message);
+                returnVal = -1;
+            }
+            return 100;
```
Comment (adityamandaleeka) [inline tests/src/JIT/Regression/JitBlue/GitHub_19288/GitHub_19288.cs]: return returnVal?

Comment (CarolEidt) [reply tests/src/JIT/Regression/JitBlue/GitHub_19288/GitHub_19288.cs]: Yes, thanks.

**Comment by CarolEidt** [comment]
@adityamandaleeka thanks for reviewing the test (your test, actually). I was hoping to see it fail on arm64 before I add the fix to the PR. I think I've got a fix, but there are a number of diffs so I'll have to take a look at them to make sure the fix is correct.

**Comment by adityamandaleeka** [comment]
Makes sense. BTW, I think you'll need to update tests/arm64/Tests.lst to include this new test and also remove the line in your csproj file setting the test priority to 1 before it'll run in the CI job.

**Comment by CarolEidt** [comment]
@dotnet-bot test Windows_NT x64 Release CoreFX Tests

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib PTAL - I'm still running diffs to assess impact beyond the new test.
I originally just pushed the new test, and verified that it fails on arm64 without this fix.

**Pull request merged**